auth/api/iam/openid4vp.go

```diff
@@ -158,6 +163,133 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 	}, nil
 }
 
+func (r Wrapper) handleAuthorizeRequestFromVerifier(ctx context.Context, walletDID did.DID, params map[string]string) (HandleAuthorizeRequestResponseObject, error) {
+	// we expect an OpenID4VP request like this
+	// GET /iam/456/authorize?response_type=vp_token&client_id=did:web:example.com:iam:123&nonce=xyz
+	//        &response_mode=direct_post&response_uri=https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb&presentation_definition_uri=example.com%2Fiam%2F123%2Fpresentation_definition?scope=a+b HTTP/1.1
+	//    Host: server.com
+	// The following parameters are expected
+	// response_type, REQUIRED.  Value MUST be set to "vp_token".
+	// client_id, REQUIRED. This must be a did:web
+	// response_uri, REQUIRED. This must be the verifier node url
+	// response_mode, REQUIRED. Value MUST be "direct_post"
+	// presentation_definition_uri, REQUIRED. For getting the presentation definition
+
+	// there are way more error conditions that listed at: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#name-error-response
+	// missing or invalid parameters are all mapped to invalid_request
+	// any operation that fails is mapped to server_error, this includes unreachable or broken backends.
+
+	responseMode, ok := params[responseModeParam]
+	if !ok || responseMode != responseModeDirectPost {
+		return nil, oauthError(oauth.InvalidRequest, "invalid response_mode parameter")
+	}
```

Comment:
Don't need `ok`, string will just be empty and check will still fail.

```suggestion
	if params[responseModeParam] != responseModeDirectPost {
		return nil, oauthError(oauth.InvalidRequest, "invalid response_mode parameter")
	}
```

This also applies to most (all?) other checks below on `params`

Reply:
I removed most of the double checks, some error messages will be more cryptic but this will mostly happen during development.